File: client-app/src/app/models/blog.ts
import { Category } from "./category";

export interface Blog {
    id: string;
    title: string;
    description: string;
    createdDate: Date;
    category: Category;
    content: string;
    image: string;
    imageId: string;
}

export class Blog implements Blog {
    id = "";
    title = "";
    description = "";
    createdDate: Date = new Date();
    category: Category = new Category();
    content = "";
    image = "";
    imageId = "";
    constructor(blog?: Blog) {
        if (blog) {
            Object.assign(this, blog);
        }
    }
}

File: Domain/Blog.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Domain
{
    public class Blog
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public DateTime CreatedDate { get; set; }
        public Category Category { get; set; }
        public string Content { get; set; }
        public string ImageId { get; set; }
        public string Image { get; set; }

    }
}